Source/driver/parse_castro_params.py

```diff
@@ -255,20 +266,25 @@ def write_headers(params, out_directory, struct_name):
         params_nm = [q for q in params if q.namespace == nm]
         # sort by repr since None may be present
         ifdefs = sorted({q.ifdef for q in params_nm}, key=repr)
-
         sf.write(f"struct {nm}_t {{\n")
         print("namespace = ", nm)
         for ifdef in ifdefs:
+            params_if = [q for q in params_nm if q.ifdef == ifdef]
+            types = set(q.dtype for q in params_if)
```

Comment:
`set` ordering is non-deterministic for strings:
```suggestion
            types = sorted(set(q.dtype for q in params_if))
```